woocommerce/woocommerce-admin
# Redirect instead of stalling on WCPay inbox note install action

Fixes https://github.com/woocommerce/woocommerce-admin/issues/5412

Switches the WCPay inbox note **from no URL** – and performing plugin installation on the async action hook, resulting in a busy state that doesn't resolve – **to an `action` param** admin URL, following (plugin _de_-activation) precedent in https://github.com/woocommerce/woocommerce-admin/pull/3687.

This allows redirecting to the next setup step after allowing installation & activation to complete, which is not possible to coordinate from the hook with a note URL alone. Adding a ToS message enables navigating directly into the WCPay onboarding flow.

Builds on earlier commits written and tested by:
- @jrodger in https://github.com/woocommerce/woocommerce-admin/pull/5361 (handling action URL param with redirect)
- @vbelolapotkov in https://github.com/woocommerce/woocommerce-admin/pull/5362 (enhancing existing note class with next step)

Open to hearing reservations (raised in https://github.com/woocommerce/woocommerce-admin/pull/5361 description) about activating a plugin from a (nonce-less) GET URL, and whether it would be better to at least wait for the note status to change (to "actioned", triggered by the parallel AJAX request) before proceeding with the installation.

_Edit:_ Also, is there a better way to be dealing with errors in the handler? Currently it will go to the home screen with the URL param still there.

### Screenshots

![wcpay-note-redirect-to-setup](https://user-images.githubusercontent.com/1867547/96268352-f4b5d380-0f96-11eb-9d8d-4c897af6f08c.gif)

### Detailed test instructions:

- Add WCPay Inbox note. I used [this change](https://github.com/woocommerce/woocommerce-admin/commit/17d141504f02c43e7721892f12fd5c6dff4f5f48) to force the note to show up, but if the note actions already exist with no URL, the note will need to be deleted first, either directly in the `wp_wc_admin_notes` table or with `Notes::delete_notes_with_name( 'wc-admin-woocommerce-payments' );`
- Take the "Get started" action on the Inbox note
- Verify that the WooCommerce Payments plugin is installed & activated, and that it redirects to the stripe.com account onboarding flow

### Changelog Note:


Fix: Avoid stalling after WCPay installation from Inbox note by facilitating setup process
